lib/src/hydrated.dart

```diff
@@ -58,48 +46,59 @@ typedef HydrateCallback<T> = T Function(String);
 ///   );
 /// ```
 class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
-  static final _areTypesEqual = TypeUtils.areTypesEqual;
   final BehaviorSubject<T> _subject;
-  final String _key;
+
+  @deprecated
+  // ignore: unused_field
   final HydrateCallback<T>? _hydrate;
+
+  @deprecated
+  // ignore: unused_field
   final PersistCallback<T>? _persist;
+
   final VoidCallback? _onHydrate;
   final T? _seedValue;
 
+  final GenericValuePersistence<T> _persistence;
+
   HydratedSubject._(
-    this._key,
     this._seedValue,
     this._hydrate,
     this._persist,
     this._onHydrate,
     this._subject,
+    this._persistence,
   ) : super(_subject, _subject.stream) {
     _hydrateSubject();
   }
 
-  factory HydratedSubject(
-    String key, {
+  factory HydratedSubject({
+    @Deprecated('`key` is deprecated. '
+        'Please use `persistence` for managing storage location.')
+        String? key,
     T? seedValue,
-    HydrateCallback<T>? hydrate,
-    PersistCallback<T>? persist,
+    @Deprecated('`hydrate` callback is deprecated. '
+        'Please use `persistence` for storing structures.')
+        HydrateCallback<T>? hydrate,
+    @Deprecated('`persist` callback is deprecated. '
+        'Please use `persistence` for storing structures.')
+        PersistCallback<T>? persist,
```

Comment:
Encoding and decoding values when persisting them is now a responsibility of the `GenericValuePersistence`.